tests/core/consensus/test_clique_consensus.py

```diff
@@ -0,0 +1,453 @@
+    return True if vote is not None else False
+
+
+def make_next_header(previous_header,
+                     signer_private_key,
+                     coinbase=ZERO_ADDRESS,
+                     nonce=NONCE_DROP,
+                     difficulty=2):
+    next_header = sign_block_header(BlockHeader.from_parent(
+        coinbase=coinbase,
+        nonce=nonce,
+        parent=previous_header,
+        timestamp=previous_header.timestamp + 1,
+        gas_limit=previous_header.gas_limit,
+        difficulty=difficulty,
+        # FIXME: I think our sign_block_header is wrong
+        extra_data=VANITY_LENGTH * b'0' + SIGNATURE_LENGTH * b'0',
+    ), signer_private_key)
+    return next_header
+
+
+@to_tuple
+def alice_nominates_bob_and_ron_then_they_kick_her():
+    header = PARAGON_GENESIS_HEADER
+
+    header = make_next_header(header, ALICE_PK)
+
+    yield header
+
+    header = make_next_header(header, ALICE_PK, BOB, NONCE_AUTH)
+    yield header
+    # At this point, we have a new signer because it just needed a single vote to win at this point
+
+    header = make_next_header(header, BOB_PK, RON, NONCE_AUTH)
+    yield header
+
+    header = make_next_header(header, ALICE_PK, RON, NONCE_AUTH)
+    yield header
+    # But we needed two votes to get a third signer approvoved (+ 50 % signers)
+
+    header = make_next_header(header, BOB_PK, ALICE, NONCE_DROP)
+    yield header
+
+    header = make_next_header(header, RON_PK, ALICE, NONCE_DROP)
+    yield header
+
+    header = make_next_header(header, BOB_PK)
+    yield header
+
+
+def validate_seal_and_get_snapshot(clique, header):
+    clique.validate_seal(header)
+    return clique.get_snapshot(header)
+
+
+@pytest.fixture
+def clique(paragon_chain_with_clique):
+    _, clique = paragon_chain_with_clique
+    return clique
+
+
+@pytest.fixture
+def paragon_chain(paragon_chain_with_clique):
+    chain, _ = paragon_chain_with_clique
+    return chain
+
+
+@pytest.fixture
+def paragon_chain_with_clique(base_db):
+
+    vms = ((0, PetersburgVM,),)
+
+    clique = CliqueConsensus(base_db)
+
+    vms = clique.amend_vm_configuration(vms)
+
+    chain = MiningChain.configure(
+        vm_configuration=vms,
+        chain_id=5,
+    ).from_genesis(base_db, PARAGON_GENESIS_PARAMS, PARAGON_GENESIS_STATE)
+    return chain, clique
+
+
+def test_can_retrieve_root_snapshot(paragon_chain, clique):
+    head = paragon_chain.get_canonical_head()
+    snapshot = clique.get_snapshot(head)
+    assert snapshot.get_sorted_signers() == [ALICE]
+
+
+def test_raises_unknown_ancestor_error(paragon_chain, clique):
+    head = paragon_chain.get_canonical_head()
+    next_header = make_next_header(head, ALICE_PK, RON, NONCE_AUTH)
+
+    with pytest.raises(ValidationError, match='Unknown ancestor'):
+        clique.get_snapshot(next_header)
+
+
+def test_import_block(paragon_chain, clique):
+
+    vm = paragon_chain.get_vm()
+    from tests.core.helpers import new_transaction
```

Comment:
>I would advocate avoiding writing new code that is dependent on `helpers`

Mmh, in that case, how about moving the `new_transaction` utility somewhere into `eth.tools`. Because I think there's value in using a real transaction here instead of a mocked one. E.g. the real one is produced based on the actual VM (so taking the transaction class of the VM into account) and will also read the nonce out of that VM.

Reply:
@cburgdorf I'm still inclined to push for factory style code rather than helper functions but :+1: for `eth.tools`